packages/utils/telemetry-utils/src/eventEmitterWithErrorHandling.ts

```diff
@@ -11,18 +11,30 @@ import { TypedEventEmitter, EventEmitterEventType } from "@fluidframework/common
  * Any exception thrown by "error" listeners will propagate to the caller.
  */
 export class EventEmitterWithErrorHandling<TEvent extends IEvent = IEvent> extends TypedEventEmitter<TEvent> {
+    constructor(private readonly errorListener?: (eventName: EventEmitterEventType, error: any) => void) {
+        super();
+        if (this.errorListener === undefined) {
+            this.errorListener = this.defaultErrorHandler.bind(this);
+        }
+    }
+
+    private defaultErrorHandler(event, error) {
+        // Some listener threw an error, we'll try emitting that error via the error event
+        // But not if we're already dealing with the error event, in that case just let the error be thrown
+        if (event === "error") {
+            throw error;
+        }
+
+        // Note: This will throw if no listeners are registered for the error event
+        super.emit("error", error);
+    }
+
     public emit(event: EventEmitterEventType, ...args: any[]): boolean {
         try {
             return super.emit(event, ...args);
         } catch (error) {
-            // Some listener threw an error, we'll try emitting that error via the error event
-            // But not if we're already dealing with the error event, in that case just let the error be thrown
-            if (event === "error") {
-                throw error;
-            }
-
-            // Note: This will throw if no listeners are registered for the error event
-            return super.emit("error", error);
+            this.errorListener!(event, error);
```

Comment:
No lint error here?  Consider decoupling the class member and constructor arg to avoid the `!`.